### PR TITLE
Facebook Authentication Refactor

### PR DESCRIPTION
**Pull request by GayathriVenkatesh**
# Description

The functions google_oauth2 and facebook are similar in functionality. This PR combines the two to reduce redundancy.

## Corresponding Issue
#1732 

Reviewing this pull request? Check out our [Code Review Practices](https://github.com/ifmeorg/ifme/wiki/Code-Review-Practices) guide if you haven't already!

### CODE REVIEW

File: app/controllers/omniauth_callbacks_controller.rb [hunk 1]
```diff
@@ -1,25 +1,22 @@
...
-    if user.present?
-      user.accept_invitation! if invitation_token
-      upload_avatar(omniauth_avatar)
-
-      flash[:notice] = I18n.t('devise.omniauth_callbacks.success',
-                              kind: t('omniauth.google'))
-      sign_in_and_redirect @user, event: :authentication
-    else
-      redirect_to new_user_session_path, notice: t('omniauth.access_denied')
+  def find_auth_translation(provider)
```
Comment (bmwachajr) [inline app/controllers/omniauth_callbacks_controller.rb:4]: we shall have to make this method private to this controller.

Even better we can move  `Google` from `omniauth:` to `navigation:` in the translations YAML. In that way we can have a neat `t('navigation.#{provider}')` this implementation scales.

File: app/controllers/omniauth_callbacks_controller.rb [hunk 2]
```diff
@@ -1,29 +1,14 @@
...
-    if user.present?
-      user.accept_invitation! if invitation_token
-      upload_avatar(omniauth_avatar)
-      flash[:notice] = I18n.t('devise.omniauth_callbacks.success',
-                              kind: 'Facebook')
-      sign_in_and_redirect @user, event: :authentication
-    else
-      redirect_to new_user_session_path, notice: t('omniauth.access_denied')
-    end
+  def omniauth_login(provider)
```
Comment (julianguyen) [inline app/controllers/omniauth_callbacks_controller.rb:4]: You'll need to update the tests since `google_oauth2` no longer exists. That's why all those tests are failing in the CircleCI build. 

I really recommend running tests locally and not depending on CircleCI.

File: app/controllers/omniauth_callbacks_controller.rb [hunk 3]
```diff
@@ -1,29 +1,14 @@
...
-      upload_avatar(omniauth_avatar)
-      flash[:notice] = I18n.t('devise.omniauth_callbacks.success',
-                              kind: 'Facebook')
-      sign_in_and_redirect @user, event: :authentication
-    else
-      redirect_to new_user_session_path, notice: t('omniauth.access_denied')
-    end
+  def omniauth_login
+    redirect_to new_user_session_path, notice:
+      t('omniauth.access_denied') if user.blank?
```
Comment (bmwachajr) [inline app/controllers/omniauth_callbacks_controller.rb:6]: redirect doesn't stop the rest of this method from executing. we need to add 'and return' to this line.

File: spec/controllers/omniauth_callbacks_controller_spec.rb [hunk 4]
```diff
@@ -140,7 +140,7 @@
     context 'when facebook email already exist in the system' do
       let!(:user) { create(:user, email: 'example@xyze.it') }
 
-      before { get :facebook }
+      before { get :omniauth_login, params: { id: 1, provider: 'facebook' } }
```
Comment (bmwachajr) [inline spec/controllers/omniauth_callbacks_controller_spec.rb:143]: I think this is a redundant param `id:`

File: spec/controllers/omniauth_callbacks_controller_spec.rb [hunk 5]
```diff
@@ -194,13 +194,13 @@
       let(:oauth_email) { request.env['omniauth.auth']['info']['email'] }
       let(:oauth_user) { User.find_by(email: oauth_email) }
       let(:user) { oauth_user }
-      before { get :facebook }
-      
+      before { get :omniauth_login }
```
Comment (bmwachajr) [inline spec/controllers/omniauth_callbacks_controller_spec.rb:197]: This action might be redundant since it is lacking params.

**Comment by bmwachajr** [comment]
@julianguyen looks like we have a couple of ruby advisories flagged by CircleCi. I am wondering whether @dependabot-bot is active!

**Comment by julianguyen** [comment]
@bmwachajr It is active, but I'm not seeing any flags about these packages. I would go ahead and update them in this PR or in another one.

**Comment by julianguyen** [comment]
Actually I'm just going to make a PR right now for these vulnerable gems and have it merged. So you'll just need to rebase.

File: config/locales/de.yml [hunk 6]
```diff
@@ -30,6 +30,7 @@ de:
     hi: 'हिंदी'
     zh-CN: 中文
   navigation:
+    google: Google
```
Comment (julianguyen) [inline config/locales/de.yml:33]: I'm curious why we are moving this Google key. It seem unnecessary. Can we undo these changes?
